Source/WebCore/workers/service/ServiceWorkerRegistration.cpp

```diff
@@ -303,9 +303,6 @@ void ServiceWorkerRegistration::showNotification(ScriptExecutionContext& context
         return;
     }
 
-    if (context.isServiceWorkerGlobalScope())
-        downcast<ServiceWorkerGlobalScope>(context).setHasPendingSilentPushEvent(false);
-
```

Comment:
I think we originally put this check here defensively so that the web developer gets credit for trying to show a notification even if creating the notification creation fails for some reason. But looking through `Notification::createForServiceWorker`, this basically shouldn't happen except in edge cases (e.g. if the web dev tries to pass in a too-large `data` property into the Notification constructor). So I guess it's okay to move this down a few lines to where you've moved it.

Reply:
Right, I think we should move it, see for instance rdar://107461729